### ISSUE TITLE
plot.mgcv.smooth.1d conditional density artefacts

### ISSUE THREAD

**Issue by mfasiolo** [mfasiolo/mgcViz#16]
Look at the left of the plot. Strangely this does not seem to happen for different seeds.

```
set.seed(414)
library(mgcViz)
n  <- 1e6
x1 <- rnorm(n)
x2 <- rnorm(n)
dat <- data.frame("x1" = x1, "x2" = x2,
                  "y" = sin(x1) + 0.5 * x2^2 + pmax(x2, 0.2) * rnorm(n))
b <- bam(y ~ s(x1)+s(x2), data = dat, method = "fREML", discrete = TRUE)
v <- getViz(b)
plot(v(2), resDen = "cond", alpha.rug = 0.05, maxpo = 1e4)
```

**Comment by ghost** [comment]
I have observed similar problems on other data set. No idea right now.

**Comment by mfasiolo** [comment]
It turns out that in this line

 ```
       estXY$fhat <- estXY$fhat / 
                      ( bkde(.datR[ , 1], gridsize = args.dens.opts$ngr[1], 
                             range.x = P$xlim, bandwidth = bw[1])$y  )
```
the conditional density in the denominator can be almost zero. This in not quite a numerical issue, it's just due to the fact that the kernel density estimator of the marginal density of x goes to zero if no nearby observation is present. Solved by adding a small constant 

`tmp <- 1e-8 / sqrt(2*pi*var(.datR[ , 1]))`

which has the nice property of taking into account the scale of x (via var(datR)). But this solution might not be bullet proof. Similar issue appeared in cdCheck. Both addressed in 4747af1c317d2bc590c2f0846b5c3b63d5f507ed
